Automattic/wp-calypso
# Patterns: add auth modal

Related to https://github.com/Automattic/dotcom-forge/issues/6075

## Proposed Changes
Logged-out users will only be able to copy the first N number of patterns. The other patterns will display a "sign in / up" buttons. With this PR we are adding a modal with these buttons.

## Testing Instructions
1) Log out from your account
2) Open `http://calypso.localhost:3000/ru/patterns/contact`
3) Command + F -> `Get access` and click on it
4) Assert that you see the next dialog <br /> ![Screenshot 2024-03-19 at 13 07 48](https://github.com/Automattic/wp-calypso/assets/5598437/fc36e96c-eb1b-4545-b879-6eb86aa2585b)
5) Assert that responsive version works good and looks according to the design BnTfb6hzKcRTb2LBtuPwKe-fi-698_4359
6) Assert that buttons works as expected: `Create a free account`,`Log in`, `X`.
